application.js

```diff
@@ -0,0 +1,253 @@
+    updateCountText()
+}
+
+// 할 일 삭제 이벤트 처리기.
+function removeCurrentTodoElement(event){
+    // 삭제된 할 일을 할 일 목록에서 제거.
+    removedTodoName = event.target.parentNode.querySelector('label').innerText
+    todoElementsNameArray.splice(todoElementsNameArray.indexOf(removedTodoName), 1)
+    // localStorage에 저장된 할 일 데이터를 지우고 할 일 목록을 업데이트.
+    localStorage.removeItem(removedTodoName)
+    localStorage.setItem(KEYWORD, JSON.stringify(todoElementsNameArray))
+    // 실제로 HTML 요소를 삭제하고 카운터 업데이트.
+    // event.target.parentNode.parentNode.remove()
+    event.target.closest('li').remove()
+    updateCountText()
+}
+
+// 할 일 변경 이벤트 처리기.
+function updateTodoEdit(event){
+    todoElementLI = event.target.parentNode
+    // ESC를 눌렀다면 편집 모드 종료, Enter를 눌렀다면 편집 적용.
+    if(event.key == 'Escape'){
+        todoElementLI.classList.toggle('editing')
+    } else if (event.key == 'Enter'){
+        // 각각 변경된 할 일 텍스트, 원래 할 일 텍스트.
+        newTodoText = event.target.value.trimStart().trimEnd()
+        updatedTodoText = todoElementLI.querySelector('div label').innerText
+        
+        // 입력값 필터링.
+        if(newTodoText.length == 0){
+            event.target.focus()
+        }
+        is_dup = false
+        todoElementsNameArray.forEach(element => {
+            if(element == newTodoText){
+                is_dup = true
+            }
+        })
+        if(is_dup){
+            alert('That ToDo is already exist!')
+            return;
+        }
+
+        // localStorage에서 해당 할 일의 완료 여부를 가져와서 따로 저장 후 해당 할 일 삭제.
+        status = localStorage.getItem(updatedTodoText)
+        localStorage.removeItem(updatedTodoText)
+        // localStorage에 변경된 할 일의 텍스트로 새로운 항목 저장. 위에서 따로 저장해둔 완료 여부를 적용.
+        localStorage.setItem(newTodoText, status)
+        // 할 일 목록에서도 변경사항 적용 및 localStorage에 반영.
+        todoElementsNameArray.splice(todoElementsNameArray.indexOf(updatedTodoText), 1, newTodoText)
+        localStorage.setItem(KEYWORD, JSON.stringify(todoElementsNameArray))
+        // HTML 요소에서도 변경사항 적용.
+        todoElementLI.querySelector('div label').innerText = newTodoText
+        todoElementLI.classList.toggle('editing')
+    }
+}
+
+function updateView(){
+    // ??? 구현 예정.
+}
+
+// 할 일이 몇 개 있는지 출력하는 텍스트(총 n 개)를 업데이트하는 로직.
+function updateCountText(){
+    todoListCountText = document.querySelector('span.todo-count strong')
+    switch(selectedFilter){
+        case filterAll:
+            todoListCountText.innerText = todoList.querySelectorAll('li').length
+            break
+        case filterActive:
+            todoListCountText.innerText = todoList.querySelectorAll('li').length - todoList.querySelectorAll('li.completed').length
+            break
+        case filterCompleted:
+            todoListCountText.innerText = todoList.querySelectorAll('li.completed').length
+    }
+}
+
+// 할 일을 더블클릭 했을 때 편집 모드 토글 로직.
+function toggleTodoElementMode(event){
+    todoElementLI = event.target.parentNode.parentNode
+    todoElementLI.classList.toggle('editing')
+}
```

Comment:
```suggestion
function toggleTodoElementMode({ target }){
    target.closest('li').classList.toggle('editing')
}
```

구조분해할당을 통해 ```target```을 받아오고 ```parentNode``` 대신 ```closest()```을 사용하면 더 깔끔하고 가독성 좋은 코드를 만들 수 있을거 같습니다!

Reply:
구조분해할당이란 개념은 처음 들어봤는데 파이썬에서 a, b, c, d, e = [1, 2, 3, 4, 5]와 비슷한 개념이군요! 적재적소에 사용하려면 시간이 좀 걸리겠지만 이런것도 가능하다는 걸 알게 되서 다행입니다ㅋㅋㅋ

review#12 커밋에서 반영하였습니다. 감사합니다.